core/src/main/java/ai/timefold/solver/core/impl/localsearch/decider/acceptor/restart/AbstractGeometricRestartStrategy.java

```diff
@@ -0,0 +1,89 @@
+package ai.timefold.solver.core.impl.localsearch.decider.acceptor.restart;
+
+import java.time.Clock;
+
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchMoveScope;
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchPhaseScope;
+import ai.timefold.solver.core.impl.solver.scope.SolverScope;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Restart strategy, which exponentially increases the metric that triggers the restart process.
+ * The first restart occurs after the {@code grace period + 1 * scalingFactor} metric.
+ * Following that, the metric increases exponentially: 1, 2, 3, 5, 7, 10, 14...
+ * <p>
+ * The strategy is based on the work: Search in a Small World by Toby Walsh
+ * 
+ * @param <Solution_> the solution type
+ */
+public abstract class AbstractGeometricRestartStrategy<Solution_> implements RestartStrategy<Solution_> {
+    private static final double GEOMETRIC_FACTOR = 1.4; // Value extracted from the cited paper
+    protected final Clock clock;
+    protected final Logger logger = LoggerFactory.getLogger(AbstractGeometricRestartStrategy.class);
+    protected final double scalingFactor;
+
+    private boolean gracePeriodFinished;
+    private boolean restartTriggered;
+    private long gracePeriodMillis;
+    protected long nextRestart;
+    protected double currentGeometricGrowFactor;
+
+    protected AbstractGeometricRestartStrategy(Clock clock, double scalingFactor) {
+        this.clock = clock;
+        this.scalingFactor = scalingFactor;
+    }
+
+    @Override
+    public void phaseStarted(LocalSearchPhaseScope<Solution_> phaseScope) {
+        restartTriggered = false;
+        if (gracePeriodMillis == 0) {
+            // 10 seconds of grace period
+            gracePeriodMillis = clock.millis() + 10_000;
```

Comment:
The [Javadoc of Clock](https://docs.oracle.com/javase/8/docs/api/java/time/Clock.html#millis--) recommends to not use `millis()`. Arguably, it'd be simpler if this code operated on `Instant` anyway - `long` brings no semantics, `Instant` does.

Reply:
Also, the magic constant should become an actual constant, and be Javadoc'd. Why is the value what it is?